neqo-transport/src/connection/mod.rs

```diff
@@ -2252,6 +2252,18 @@ impl Connection {
             let payload_start = builder.len();
             let (mut tokens, mut ack_eliciting, mut padded) = (Vec::new(), false, false);
             if let Some(ref close) = closing_frame {
+                if builder.remaining() > ClosingFrame::min_length() + RecvdPackets::max_len() {
```

Comment:
This little chunk of code (under `if let Some(ref close) ...` is large enough now that it probably needs its own function.

As for this line of code, I think that what you want to do is ask the closing frame how much space it is going to take up, then - if there is enough space for a minimal ACK, which might assume ECN for the sake of simplicity - you would temporarily reduce "remaining" by the closing frame size, write the ACK, add back the closing frame size, and off you go.

The close frame size computation doesn't need to take "remaining" into account.  We just need to know how big it would prefer to be.